### PR TITLE
Add KnpSnappyWriter

### PR DESCRIPTION
**Pull request by greg0ire**
### Changelog

```markdown
### Added
- Added pdf writer based on [knp/snappy](https://packagist.org/packages/knplabs/knp-snappy)
```

### Subject

This PR is my take on #76 , thanks @chalasr for what you already did

### To do

~~- [ ] consider relying on knp/snappy-bundle to inject the dependency~~ this is not a symfony bundle so we can't
- [x] fix DIP violation (dependency on Pdf is not injected)
- [x] write real unit tests (the tests that already exist are integration tests). Implies applying this : https://phpunit.de/manual/current/en/test-doubles.html#test-doubles.mocking-the-filesystem
- [x] document usage

### CODE REVIEW

File: docs/reference/outputs.rst [hunk 1]
```diff
@@ -12,5 +12,33 @@ Several output formatters are supported:
...
+    <?php
+
+    use Exporter\Writer\KnpSnappyWriter;
+    use Knp\Snappy\Pdf;
+
+    $writer = new KnpSnappyWriter(
+        new Pdf(),
+        'output.pdf',
+        '/absolute/path/to/the/wkhtmltopdf/binary',
+        true // include a header
```
Comment (core23) [inline docs/reference/outputs.rst]: Commas are missing

File: lib/Exporter/Writer/KnpSnappyWriter.php [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    </head>
+    <body>
+        [header-html]
+        <table>[content]</table>
+        [footer-html]
+    </body>
+</html>';
+
+    /**
+     * @throws \RuntimeException
```
Comment (core23) [inline lib/Exporter/Writer/KnpSnappyWriter.php]: What about `InvalidArgumentException`?

File: lib/Exporter/Writer/KnpSnappyWriter.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+                gettype($showBorders)
+            ));
+        }
+
+        if (is_file($filename)) {
+            throw new \RuntimeException(sprintf('The file "%s" already exists', $filename));
+        }
+
+        if (!is_executable($wkhtmltopdf)) {
+            throw new \RuntimeException('The file "%s" doesn\'t exist or is not executable', $wkhtmltopdf);
```
Comment (greg0ire) [inline lib/Exporter/Writer/KnpSnappyWriter.php]: @chalasr missing `sprintf` here ;)

File: lib/Exporter/Writer/KnpSnappyWriter.php (hunk 2 above)
Comment (greg0ire) [inline lib/Exporter/Writer/KnpSnappyWriter.php]: Good catch!

File: lib/Exporter/Writer/KnpSnappyWriter.php (hunk 3 above)
Comment (chalasr) [inline lib/Exporter/Writer/KnpSnappyWriter.php]: Oups... :)

**Comment by greg0ire** [comment]
Woops, I accidentally pushed the branch in the main repo instead of my fork…

**Pull request closed without merging**